modules/git/repo_attribute.go

```diff
@@ -134,11 +121,21 @@ func (c *CheckAttributeReader) Init(ctx context.Context) error {
 		return fmt.Errorf("no provided Attributes to check")
 	}
 
-	cmdArgs = append(cmdArgs, c.Attributes...)
-	cmdArgs = append(cmdArgs, "--")
-
 	c.ctx, c.cancel = context.WithCancel(ctx)
-	c.cmd = NewCommand(c.ctx, cmdArgs...)
+	c.cmd = NewCommand(c.ctx, "check-attr", "--stdin", "-z")
+
+	if len(c.IndexFile) > 0 {
+		c.cmd.AddArguments("--cached")
+		c.env = append(c.env, "GIT_INDEX_FILE="+c.IndexFile)
+	}
+
+	if len(c.WorkTree) > 0 {
+		c.env = append(c.env, "GIT_WORK_TREE="+c.WorkTree)
+	}
+
+	c.env = append(c.env, "GIT_FLUSH=1")
+
+	c.cmd.AddDynamicArguments(c.Attributes...).AddArguments("--")
```

Comment:
```suggestion
	c.cmd.AddDynamicArguments(c.Attributes...).AddArguments("--") // Files will be added later on
```

Reply:
Not related, I do not want to touch old code. And I do not see why there would be files to be added there.

There just seems something incorrect. So the comment doesn't seem to be proper either.

Reply:
If you read code, you can see that after this line, the next usage of `c.cmd` is `c.cmd.Run()`

So, no more argument would be added.

I have noticed this empty `--` in the first refactoring PR, but I always prefer to leave it as it is, to reduce unnecessary logic changes.


![image](https://user-images.githubusercontent.com/2114189/216737333-5489062d-7ba0-4ebb-b992-a10590d35de0.png)

Reply:
Added comment:

// The empty "--" comes from #16773 , and it seems unnecessary because nothing else would be added later.